File: Typescript/test/destCity.test.ts
import { expect, jest, test } from '@jest/globals';
import * as solution from '../src/destCity';

test('destCity Example 1', () => {
	let input0 = [["London", "New York"], ["New York", "Lima"], ["Lima", "Sao Paulo"]];
	let expected = "Sao Paulo";
	expect(solution.destCity(input0)).toBe(expected);
});

test('destCity Example 2', () => {
	let input0 = [["B", "C"], ["D", "B"], ["C", "A"]];
	let expected = "A";
	expect(solution.destCity(input0)).toBe(expected);
});

test('destCity Example 3', () => {
	let input0 = [["A", "Z"]]
	let expected = "Z";
	expect(solution.destCity(input0)).toBe(expected);
});

File: Dart/test/destCity_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/destCity.dart';

void main() {
  test("test destCity", () {
    var input0 = [
      [
        ["London", "New York"],
        ["New York", "Lima"],
        ["Lima", "Sao Paulo"]
      ],
      [
        ["B", "C"],
        ["D", "B"],
        ["C", "A"]
      ],
      [
        ["A", "Z"]
      ]
    ];
    var expected = ["Sao Paulo", "A", "Z"];

    final sln = Solution();
    for (var i = 0; i < input0.length; i++) {
      var output = sln.destCity(input0[i]);
      expect(output, expected[i]);
    }
  });
}
